### PR TITLE
feat: IMPLEMENTS CHECKING FOR VARIABLE MATES

### PR DESCRIPTION
**Pull request by fraserwg**
On a latlon LLC grid some variables have mates. This commit will produce a meaningful error if attempting to use `get_dataset` or `faces_dataset_to_latlon` to load a variable without its mate.

See Github issue #234

### CODE REVIEW

File: xmitgcm/llcreader/llcmodel.py [hunk 1]
```diff
@@ -349,13 +357,21 @@ def _all_facets_to_latlon(data_facets, meta):
...
+            mate = None
+
+        # Raises an exception if the mate of a variable in vnames is missing.
+        if mate is not None:
+            vector_pairs.append((vname, mate))
+            try:
+                vnames.remove(mate)
+            except ValueError:
+                msg = 'If {} in varnames, {} must also be in varnames'.format(vname, mate)
+                raise ValueError(msg)
```
Comment (rabernat) [inline xmitgcm/llcreader/llcmodel.py:374]: This PR makes me realize we have some duplicative code paths in here. I'm trying to understand why we need to implement this in both `_all_facets_to_latlon` and `faces_dataset_to_latlon`. I think that `faces_dataset_to_latlon` should probably be refactored to call `_all_facets_to_latlon`. However, that is not your problem.

**Review by rabernat — CHANGES_REQUESTED** [review]
This looks great! Thank you so much.

We will just need a test for this scenario. It would go in here: https://github.com/MITgcm/xmitgcm/blob/7c5305cc81dee6a21e0b035e8682d03c113c10de/xmitgcm/test/test_llcreader.py#L76

Something like
```python
def test_vector_mate_error(local_llc90_store):
    store = local_llc90_store
    model = llcreader.LLC90Model(store)
    with pyest.raises(ValueError, match=r".* must also be .*"):
        ds_latlon = model.get_dataset(type='latlon', varnames=['U'], iter_start=0, iter_stop=9, iter_step=8)
```

**Comment by fraserwg** [comment]
I've added the test function but am now getting test fails on the Xarray-master build, details below:

```python
=========================== short test summary info ============================
257
FAILED xmitgcm/test/test_llcreader.py::test_ecco_portal_load[2160] - ValueErr...
258
FAILED xmitgcm/test/test_llcreader.py::test_ecco_portal_load[4320] - ValueErr...
259
====== 2 failed, 276 passed, 3 skipped, 124 warnings in 106.82s (0:01:46) ======
```

The tests weren't failing for the previous commit and I haven't changed the tests that are now failing. Also the other builds are passing (as are the tests on my local machine) -- could this be a problem with a new commit on the Xarray-master branch? Any ideas for where to start with debugging would be appreciated, I'm relatively new to the world of CI and pytest!

**Comment by fraserwg** [comment]
Have just added a new commit with the test removed again. Looks like everything's passed suggesting the problem is indeed with my test function.

**Comment by fraserwg** [comment]
Ok, all the problems have been sorted, it was just a missing pair of brackets :)

**Comment by rabernat** [comment]
Fantastic! Thanks so much. I appreciate your patience to sort out the CI failures.

**Pull request merged**